crmsh/parallax.py

```diff
@@ -1,110 +1,31 @@
-            src=None, dst=None, askpass=False, ssh_options=None, strict=True):
-        self.nodes = nodes
-        self.askpass = askpass
-        self.ssh_options = ssh_options
-        self._sudoer = None
-        self.strict = strict
+from crmsh.prun import prun
 
-        # used for call
-        self.cmd = cmd
-        # used for slurp
-        self.localdir = localdir
-        self.filename = filename
-        # used for copy
-        self.src = src
-        self.dst = dst
 
-        self.opts = self.prepare()
+Error = prun.PRunError
 
-    def prepare(self):
-        opts = parallax.Options()
-        if self.ssh_options is None:
-            self.ssh_options = ['StrictHostKeyChecking=no',
-                    'ConnectTimeout=10',
-                    'LogLevel=error']
-        sudoer, _ = crmsh.utils.user_pair_for_ssh(self.nodes[0])
-        if sudoer is not None:
-            # FIXME: this is really unreliable
-            self.ssh_options.append('IdentityFile={}/.ssh/id_rsa'.format(userdir.gethomedir(sudoer)))
-            self._sudoer = sudoer
-        opts.ssh_options = self.ssh_options
-        opts.askpass = self.askpass
-        # warn_message will available from parallax-1.0.5
-        if hasattr(opts, 'warn_message'):
-            opts.warn_message = False
-        opts.localdir = self.localdir
-        return opts
 
-    def handle(self, results):
-        for host, result in results:
-            if isinstance(result, parallax.Error) and self.strict:
-                raise ValueError("Failed on {}: {}".format(host, result))
-        return results
-
-    def call(self):
-        host_port_user = []
-        for host in self.nodes:
-            _, remote_user = crmsh.utils.user_pair_for_ssh(host)
-            host_port_user.append([host, None, remote_user])
-        # FIXME: this is really unreliable
-        sudoer = userdir.get_sudoer()
-        cmd = f'sudo bash -c "{self.cmd}"' if sudoer else self.cmd
-        results = parallax.call(host_port_user, cmd, self.opts)
-        return self.handle(list(results.items()))
-
-    def slurp(self):
-        dst = os.path.basename(self.filename)
-        results = parallax.slurp(self.nodes, self.filename, dst, self.opts)
-        return self.handle(list(results.items()))
-
-    def copy(self):
-        results = parallax.copy(self.nodes, self.src, self.dst, self.opts)
-        return self.handle(list(results.items()))
-    def run(self):
-        sudoer = userdir.get_sudoer()
-        cmd = f'sudo bash -c "{self.cmd}"' if sudoer else self.cmd
-        host_port_user = []
-        for host in self.nodes:
-            _, remote_user = crmsh.utils.user_pair_for_ssh(host)
-            host_port_user.append([host, None, remote_user])
-        return parallax.run(host_port_user, cmd, self.opts)
-
-
-def parallax_call(nodes, cmd, askpass=False, ssh_options=None, strict=True):
+def parallax_call(nodes, cmd, *, timeout_seconds: int = -1):
```

Comment:
What does this `*` mean?

Reply:
It is PEP-3102 *Keyword-only Arguments*.